src/Utility.ts

```diff
@@ -17,62 +20,38 @@ export namespace Utility {
         }
     }
 
-    export async function executeCMD(outputPane: vscode.OutputChannel, command: string,
-                                     options: child_process.SpawnOptions, ...args: string[]): Promise<void> {
+    export async function executeCMD(outputPane: OutputChannel, command: string, options: SpawnOptions, ...args: string[]): Promise<void> {
         await new Promise((resolve: () => void, reject: (e: Error) => void): void => {
             outputPane.show();
             let stderr: string = '';
-            const p: child_process.ChildProcess = child_process.spawn(command, args, options);
+            const p: ChildProcess = child_process.spawn(command, args, options);
             p.stdout.on('data', (data: string | Buffer): void =>
                 outputPane.append(data.toString()));
             p.stderr.on('data', (data: string | Buffer) => {
                 stderr = stderr.concat(data.toString());
                 outputPane.append(data.toString());
             });
             p.on('error', (err: Error) => {
-                reject(new Error(err.toString()));
+                reject(err);
             });
             p.on('exit', (code: number, signal: string) => {
                 if (code !== 0) {
-                    // tslint:disable-next-line:quotemark
-                    reject (new Error(localize('tomcatExt.commandfailed', 'Command failed with exit code {0}', code)));
-                } else {
-                    resolve();
+                    reject(new Error(localize('tomcatExt.commandfailed', 'Command failed with exit code {0}', code)));
                 }
+                resolve();
             });
         });
     }
 
-    export function isPathEqual(fsPath1: string, fsPath2: string): boolean {
-        const relativePath: string = path.relative(fsPath1, fsPath2);
-        return relativePath === '';
-    }
-
-    export function isSubPath(expectParent: string, expectChild: string): boolean {
-        const relativePath: string = path.relative(expectParent, expectChild);
-        return relativePath !== '' && !relativePath.startsWith('..') && relativePath !== expectChild;
-    }
-
-    export function getWorkspaceFolder(fsPath: string): vscode.WorkspaceFolder | undefined {
-        if (vscode.workspace.workspaceFolders) {
-           return vscode.workspace.workspaceFolders.find(
-                (f: vscode.WorkspaceFolder): boolean => {
-                return isPathEqual(f.uri.fsPath, fsPath) || isSubPath(f.uri.fsPath, fsPath);
-            });
-        } else {
+    export function getWorkspaceFolder(fsPath: string): WorkspaceFolder | undefined {
```

Comment:
I think we can use the `vscode.workspace.getWorkspaceFolder` api here?

Reply:
yes, seems there is not necessary to define a function.